.github/workflows/queasars_deployment.yml

```diff
@@ -0,0 +1,94 @@
+# This workflow runs on protected version tags. It runs the CI, checks the tag / workflow input version
+# for consistency with the pyproject.toml version and then deploys to PYPI and GitHub pages.
+
+name: QUEASARS Deployment
+
+on:
+  # Runs only on protected tags of the form vX.Y.Z
+  push:
+    tags: ['v[0-9].[0-9].[0-9]']
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version of the release in the form: vX.Y.Z'
+        type: string
+        required: true
+      deploy_to_pypi:
+        description: 'Boolean value indicating whether QUEASARS should be deployed to PyPI.'
+        type: boolean
+        required: false
+        default: true
+      deploy_to_pages:
+        description: 'Boolean value indicating whether QUEASARS documentation should be deployed to GitHub Pages.'
+        type: boolean
+        required: false
+        default: true
+
+concurrency:
+  # Only let one deployment run at a time.
+  group: "queasars deployment"
+  cancel-in-progress: false
+
+jobs:
+  Check_Version_Consistency:
+    # Check that the project version given by the tag / manual workflow input matches the pyproject.toml version
+    name: Check Version Consistency
+    runs-on: ubuntu-latest
+    steps:
+      - if: ${{ inputs.version != '' }}
+        name: Check workflow version input
+        shell: bash
+        run: |
+          if [[ $inputs.version != @(v[0-9].[0-9].[0-9]) ]]; then exit 1; fi
+          echo "version=${inputs.version:1}" >> $GITHUB_ENV
```

Comment:
This adds an environment variabel `version` for subsequent jobs, right? In that case, would it be better to call the  variable `VERSION` instead?

Reply:
Thanks!